launcher/src/main/java/net/flintmc/launcher/LaunchController.java

```diff
@@ -196,6 +168,12 @@ public void run() {
         plugin.preLaunch(rootLoader);
       }
 
+      if (launchArguments.getLaunchTarget().isEmpty()) {
+        while (true) {
+          Thread.sleep(Long.MAX_VALUE);
+        }
```

Comment:
What is the purpose of this?

Reply:
Without the infinite loop the process would just exit after the main method has been invoked

Reply:
Isn't this the wanted behavior? The process should exit after the main terminated. Why should it keep running?

Reply:
Well if there is no launch target defined nothing more will happen in the main method than loading packages and stuff like that, everything else is done by the packages.

Reply:
That doesn't really make sense though. There still will be some application to launch which contains the main method, even if this might be a package.

And even IF the packages were to do all work, the main method would either not return until all work is done or be blocked by some thread.